pymeasure/instruments/lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,117 @@
+        'CRDG? {ch}',
+        """Read the temperature in celcius from a channel."""
+    )
+    sensor = Instrument.measurement(
+        'SRDG? {ch}',
+        """Read the temperature in sensor units from a channel."""
+    )
+
+    def wait_for_temperature(self, target, unit='kelvin', accuracy=0.1,
+                             interval=1, timeout=360,
+                             should_stop=lambda: False):
+        """ Blocks the program, waiting for the temperature to reach the target
+        within the accuracy (%), checking this each interval time in seconds.
+
+        :param target: Target temperature in kelvin, celcius, or sensor units.
+        :param accuracy: An acceptable percentage deviation between the
+                         target and temperature
+
+        :param timeout: A timeout in seconds after which an exception is raised
+        :param should_stop: A function that returns True if waiting should stop, by
+                            default this always returns False
+        """
+
+        def percent_difference(temperature):
+            return np.abs(100 * (temperature - target) / target)
+
+        t = time()
+        target_acquired = False
+        while not target_acquired:
+            reading = getattr(self, unit)
+            reading = np.array([reading]) if self.id != '0' else np.array(reading)
+            diff = percent_difference(reading)
+            target_acquired = True if np.all(diff < accuracy) else False
+            sleep(interval)
+            if (time() - t) > timeout:
+                raise Exception((
+                                    "Timeout occurred after waiting %g seconds for "
+                                    "the LakeShore 331 temperature to reach %g %s."
+                                ) % (timeout, target, unit))
+            if should_stop():
+                return
+
+
+class LakeShoreHeaterOutputChannel(Channel):
+    """ Heater output channel on a lakeshore temperature controller. Provides properties to query
+    the output power in percent of the max, set the manual output power, heater range, and PID
+    temperature setpoint.
+    """
+
+    output = Instrument.measurement(
+        'HTR? {ch}',
+        """Query the heater output in percent of the max."""
+    )
+    mout = Instrument.control(
+        'MOUT? {ch}',
+        'MOUT {ch},%f',
+        """Manual heater output in percent."""
+    )
+    range = Instrument.control(
+        'RANGE? {ch}',
+        'RANGE {ch},%i',
+        """String property controlling heater range, which can take the
+       values: off, low, medium, and high.""",
+        validator=strict_discrete_set,
+        values={'off': 0, 'low': 1, 'medium': 2, 'high': 3},
+        map_values=True)
+    setpoint = Instrument.control(
+        'SETP? {ch}', 'SETP {ch},%f',
+        """A floating point property that control the setpoint temperature
+        in the preferred units of the control loop sensor."""
```

Comment:
Is there an interface to adjust the preferred unit?

Reply:
As far as I can tell, one can only set the units manually on the device itself and not through a computer interface. I haven't found any command to set the preferred units yet...

Reply:
OK, thanks, just wanted to make sure.